lib/buffer.js

```diff
@@ -67,13 +73,60 @@ function Buffer(arg, encoding) {
   return fromObject(arg);
 }
 
+/**
+ * Functionally equivalent to Buffer(arg, encoding) but throws a TypeError
+ * if value is a number.
+ **/
+Buffer.from = function(value, encoding) {
+  if (typeof value === 'number')
+    throw new TypeError(`'value' argument must not be a number`);
+  return typeof value === 'string' ?
+    fromString(value, encoding) :
+    fromObject(value);
+};
+
 Object.setPrototypeOf(Buffer.prototype, Uint8Array.prototype);
 Object.setPrototypeOf(Buffer, Uint8Array);
 
+/**
+ * Creates a new zero-filled Buffer instance.
+ **/
+Buffer.alloc = function(size, fill) {
+  if (typeof size !== 'number')
+    throw new TypeError(`'size' argument must be a number`);
+  if (fill && typeof fill !== 'string')
+    throw new TypeError(`If given, 'fill' argument must be a string`);
+  size = Math.max(0, size) || 0;
+  return fill ?
+    Buffer.allocUnsafe(size).fill(fill) :
+    createBuffer(size);
+};
+
+/**
+ * Equivalent to Buffer(num), creates a non-zero-filled Buffer instance.
+ **/
+Buffer.allocUnsafe = function(size) {
+  if (arguments.length !== 1)
+    throw new TypeError(`allocUnsafe(size) requires only a single argument`);
+  // if process.zeroFillAllBuffers is true, we must fill the buffer
+  if (process.zeroFillAllBuffers)
+    return Buffer.alloc(size);
```

Comment:
Don't do it here. Instead should set a variable in `src/node.cc` that's checkecd by `ArrayBufferAllocator::Allocate`  to always use `calloc()`. Something like this:
```cpp
void* ArrayBufferAllocator::Allocate(size_t size) {
  if (env_ == nullptr ||
      !env_->array_buffer_allocator_info()->no_zero_fill() ||
      zero_fill_all_buffers) {
    return calloc(size, 1);
  }
  env_->array_buffer_allocator_info()->reset_fill_flag();
  return malloc(size);
}
```